File: .API/Cloud/CloudMessage.cs
// Decompiled with JetBrains decompiler
// Type: CloudX.Shared.CloudMessage
// Assembly: CloudX.Shared, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
// MVID: 2635624C-5F24-4EFB-ACD1-7E9C1349B2F5
// Assembly location: F:\SteamLibrary\steamapps\common\NeosVR\HeadlessClient\CloudX.Shared.dll

using Newtonsoft.Json;
using System.Text.Json.Serialization;

namespace CloudX.Shared
{
  public class CloudMessage
  {
    [JsonProperty(PropertyName = "Message")]
    [JsonPropertyName("Message")]
    public string Message { get; set; }

    public static string ExtractMessage(string content)
    {
      try
      {
        return JsonConvert.DeserializeObject<CloudMessage>(content)?.Message ?? content;
      }
      catch
      {
        return content;
      }
    }
  }
}

File: src/api/CloudMessage.ts
export class CloudMessage {
	public Message: string;
	constructor($b: CloudMessageJSON) {
		this.Message = $b.Message;
	}
	public static ExtractMessage(content: string): string {
		try {
			return JSON.parse(content)?.Message ?? content;
		} catch (error) {
			return content;
		}
	}
	toJSON(): CloudMessageJSON {
		return { Message: this.Message };
	}
}
export interface CloudMessageJSON {
	Message: string;
}
